File: src/main/java/compiler/Parameter.java
package compiler;

public class Parameter {

	private String type;
	private String name;

	// Represents what is input for ASM. E.g. int = 'I', String = 'Ljava/lang/String;'
	private String asmType;

	public Parameter(String type, String name) {
		this.setType(type);
		this.name = name;


	}
	
	public String getName() {
		return name;
	}
	
	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	@Override
	public String toString(){
		return getType() + " : " + name;
	}

	public String getAsmType() {

		if (getType().equals("int")) {
			return  "I";
		}
		if (getType().equals("String")) {
			return "Ljava/lang/String;";
		}
		return null;
	}
}

File: src/main/scala/compiler/Parameter.scala
package compiler

// Represents what is input for ASM. E.g. int = 'I', String = 'Ljava/lang/String;'
class Parameter(val `type`: String, var name: String) {



  // Name of the variable
  def getName: String = name

  // Type of the variable
  def getType: String = `type`

  def setType(`type`: String) {
    this.`type` = `type`
  }

  // ASM type. int = "I", String = "java.lang.String", etc
  def getAsmType: String = {
    if (getType == "int") return "I"
   
    if (getType == "String") return "Ljava/lang/String;"

    return null
  }

  override def toString: String = {
    return `type` + " : " + name
  }

}
